AWSCognitoAuth/AWSCognitoAuth.m

```diff
@@ -336,9 +336,11 @@ - (void)getSessionInternal: (nullable UIViewController *) vc completion: (AWSCog
         if(expirationDate && scopes != nil && [scopes isEqualToString:[self normalizeScopes]]){
             NSDate *expiration = [self dateFromString:expirationDate];
             NSString * refreshToken = [self refreshTokenFromKeyChain:keyChainNamespace];
+            NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoAuthUserAccessToken];
+            NSString * accessToken = self.keychain[accessTokenKey];
 
             //if the session expires > 5 minutes return it.
-            if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending){
+            if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending && accessToken){
                 NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoAuthUserIdToken];
                 NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoAuthUserAccessToken];
                 AWSCognitoAuthUserSession * session = [[AWSCognitoAuthUserSession alloc] initWithIdToken:self.keychain[idTokenKey] accessToken:self.keychain[accessTokenKey] refreshToken:refreshToken expirationTime:expiration];
```

Comment:
Minor code duplication: can replace with the variables declared on 339 & 340.

Reply:
I don't understand really well. What change do you suggest?

Reply:
Created a PR for the test case - https://github.com/aws-amplify/aws-sdk-ios/pull/1940.